.github/workflows/release.yml

```diff
@@ -1,83 +1,83 @@
 name: Release
-
 on:
   workflow_dispatch:
     inputs:
       name:
-        description: 'Release name'
+        description: "Release name"
         required: true
-
 jobs:
   build:
     name: Build the release for ${{ matrix.os.runner }}
     runs-on: ${{ matrix.os.runner }}
-
     strategy:
       matrix:
         os:
           - runner: windows-latest
             exe: marksman.exe
             task: publishTo
+            rid: win-x64
           - runner: ubuntu-latest
             exe: marksman
             task: publishTo
+            rid: linux-x64
+          - runner: ubuntu-latest
+            exe: marksman
+            task: publishTo
+            rid: linux-arm64
           - runner: macos-latest
             exe: marksman
             task: macosUniversalBinary
-
+            rid: osx
     steps:
       - uses: actions/checkout@v3
-
       - name: Setup dotnet
         uses: actions/setup-dotnet@v3
         with:
-          dotnet-version: '7.0.x'
-
+          dotnet-version: "7.0.x"
       - name: Install dependencies
         run: dotnet restore
-
       - name: Build the release binary
-        run: make ${{ matrix.os.task }} DEST=out
-
+        run: make ${{ matrix.os.task }} RID=${{ matrix.os.rid }} DEST=out
       - name: Upload the binary
         uses: actions/upload-artifact@v3
         with:
-          name: ${{ matrix.os.runner }}-${{ matrix.os.exe }}
+          name: ${{ matrix.os.runner }}-${{ matrix.os.rid }}
```

Comment:
I think this is going to change the name of windows artifact to `marksman.exe-win-x64` which is not quite right: for windows we'd need either `marksman.exe` (preferably) or `marksman-x64.exe`. Also, we don't need `RID` for macos because we build universal binaries there.

I'd say it'd be easier to tweak `exe` and `task` attributes above to include `RID` where needed.

Reply:
I'll look into this tomorrow evening hopefully